kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/CheckpointInstance.java

```diff
@@ -36,19 +53,32 @@ public class CheckpointInstance
     public final long version;
     public final Optional<Integer> numParts;
 
+    public final CheckpointFormat format;
+
+    public final Optional<Path> filePath;
+
     public CheckpointInstance(String path) {
         String[] pathParts = getPathName(path).split("\\.");
+        this.filePath = Optional.of(new Path(path));
 
         if (pathParts.length == 3 && pathParts[1].equals("checkpoint") &&
                 pathParts[2].equals("parquet")) {
             // Classic checkpoint 00000000000000000010.checkpoint.parquet
             this.version = Long.parseLong(pathParts[0]);
             this.numParts = Optional.empty();
+            this.format = CheckpointFormat.SINGLE_PART;
```

Comment:
With V2 Checkpoints, a checkpoint that appears to be a single_part checkpoint can actually be a V2 checkpoint. The only way to be sure about whether the checkpoint is V2 or classic is to read the checkpoint and look for the `CheckpointManifest` option.

Reply:
Got it, fixed.